src/emu/debug/debugcpu.cpp

```diff
@@ -871,34 +871,31 @@ void debugger_cpu::reset_transient_flags()
 
 void debugger_cpu::process_source_file()
 {
-	/* loop until the file is exhausted or until we are executing again */
-	while (m_source_file != nullptr && m_execution_state == EXECUTION_STATE_STOPPED)
+	std::string buf;
+
+	// loop until the file is exhausted or until we are executing again
+	while (m_source_file && m_execution_state == EXECUTION_STATE_STOPPED)
 	{
-		/* stop at the end of file */
-		if (feof(m_source_file))
+		// stop at the end of file
+		if (m_source_file->eof())
 		{
-			fclose(m_source_file);
-			m_source_file = nullptr;
+			m_source_file.reset();
 			return;
 		}
 
-		/* fetch the next line */
-		char buf[512];
-		memset(buf, 0, sizeof(buf));
-		fgets(buf, sizeof(buf), m_source_file);
+		// fetch the next line
+		std::getline(*m_source_file, buf);
```

Comment:
You aren't checking failbit/badbit before using the result of this I/O operations.  That's bad form.  This also looks like an overly literal conversion of the C loop.  Try to make more idiomatic use of a `std::istream`.